app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedActivity.kt

```diff
@@ -17,22 +18,27 @@ class RecentlyPlayedActivity : InjectableAppCompatActivity(), RouteToExploration
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     activityComponent.inject(this)
-    val internalProfileId = intent.getIntExtra(
-      RECENTLY_PLAYED_ACTIVITY_INTERNAL_PROFILE_ID_KEY,
-      -1
+    val recentlyPlayedActivityIntentExtras = RecentlyPlayedActivityIntentExtras.parseFrom(
+      intent.getByteArrayExtra(RECENTLY_PLAYED_ACTIVITY_INTENT_EXTRAS)
     )
-    recentlyPlayedActivityPresenter.handleOnCreate(internalProfileId)
+
+    recentlyPlayedActivityPresenter.handleOnCreate(recentlyPlayedActivityIntentExtras)
   }
 
   companion object {
     // TODO(#1655): Re-restrict access to fields in tests post-Gradle.
-    const val RECENTLY_PLAYED_ACTIVITY_INTERNAL_PROFILE_ID_KEY =
-      "RecentlyPlayedActivity.internal_profile_id"
+    const val RECENTLY_PLAYED_ACTIVITY_INTENT_EXTRAS = "RecentlyPlayedActivity.intent_extras"
 
     /** Returns a new [Intent] to route to [RecentlyPlayedActivity]. */
-    fun createRecentlyPlayedActivityIntent(context: Context, internalProfileId: Int): Intent {
+    fun createRecentlyPlayedActivityIntent(
+      context: Context,
+      recentlyPlayedActivityIntentExtras: RecentlyPlayedActivityIntentExtras
+    ): Intent {
       val intent = Intent(context, RecentlyPlayedActivity::class.java)
-      intent.putExtra(RECENTLY_PLAYED_ACTIVITY_INTERNAL_PROFILE_ID_KEY, internalProfileId)
+      intent.putExtra(
```

Comment:
Prefer using this utility instead: https://github.com/oppia/oppia-android/blob/develop/utility/src/main/java/org/oppia/android/util/extensions/BundleExtensions.kt#L13.